runtime/jcl/common/reflecthelp.c

```diff
@@ -1720,3 +1720,225 @@ getFieldsHelper(JNIEnv *env, jobject cls)
 	vmFuncs->internalExitVMToJNI(vmThread);
 	return result;
 }
+
+/* Determine whether currentMethod is the accessor method for a record component with name componentName.
+ * an accessor method will have the same name as the component and take zero parameters.
+ */
+static U_8
+isRecordComponentAccessorMethodMatch(J9Method *currentMethod, const char* componentName) {
+	J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(currentMethod);
+	J9UTF8 *methodNameUtf8 = J9ROMMETHOD_NAME(romMethod);
+	U_16 methodLength = J9UTF8_LENGTH(methodNameUtf8);
+	const U_8 *methodName = J9UTF8_DATA(methodNameUtf8);
+	if ((methodLength == strlen(componentName)) 
+		&& (0 == strcmp((const char*)methodName, componentName))
+	) {
+		/* verify parameters. An accessor will not have any input parameters. */
+		J9UTF8 *methodSignatureUtf8 = J9ROMMETHOD_SIGNATURE(romMethod);
+		const U_8 *methodSignature = J9UTF8_DATA(methodSignatureUtf8);
+		if ((J9UTF8_LENGTH(methodSignatureUtf8) >= 2)
+			&& ('(' == methodSignature[0]) && (')' == methodSignature[1])
+		) {
+			return TRUE;
+		}
+	}
+	return FALSE;
+}
+
+/* Find the accessor method for a record component with name componantName. 
+ * recordComponentSlot slot for current record component will be used to optimize search
+ */
+static J9Method*
+findRecordComponentAccessorMethod(J9VMThread* currentThread, J9Class* clazz, U_32 recordComponentSlot, U_32 recordComponentCount, const char* componentName)
+{
+	J9Method *currentMethod = clazz->ramMethods;
+	U_32 romMethodCount = clazz->romClass->romMethodCount;
+	J9Method *resultMethod = NULL;
+
+	/* The compiler should generate each of the record components in order as the final recordComponentCount methods in the record class. 
```

Comment:
```suggestion
	/* javac typically generates each of the record components in order as the final recordComponentCount methods in the record class. 
```

Is this true even when there are other user methods in the record?

Reply:
yes. Custom methods are generated first, then the generated methods (toString, hashCode, equals), then the accessor methods.